cmd/gcp-controller-manager/app/node_annotater.go

```diff
@@ -287,3 +310,84 @@ func extractKubeLabels(instance *compute.Instance) (map[string]string, error) {
 
 	return labels, nil
 }
+
+// AddOrUpdateTaint tries to add a taint to taint list. Returns a new copy of updated Node and true if something was updated
+// false otherwise.
+func addOrUpdateTaint(node *core.Node, taint *core.Taint) bool {
+	nodeTaints := node.Spec.Taints
+
+	var newTaints []core.Taint
+	updated := false
+	for i := range nodeTaints {
```

Comment:
`for _, nodeTaint := range nodeTaints {...`

Reply:
https://github.com/kubernetes/cloud-provider-gcp/pull/50#discussion_r215781623

Reply:
You're not mutating the slice value below